server/plugins/TriggerUGens.cpp

```diff
@@ -895,23 +897,67 @@ void SetResetFF_next_k(SetResetFF* unit, int inNumSamples) {
 
 
 void ToggleFF_Ctor(ToggleFF* unit) {
-    SETCALC(ToggleFF_next);
+    if (unit->mCalcRate == calc_FullRate) {
+        if (INRATE(0) == calc_FullRate) {
+            SETCALC(ToggleFF_next);
+        } else {
+            SETCALC(ToggleFF_next_ak);
+        }
+    } else {
+        if (INRATE(0) == calc_FullRate) {
+            SETCALC(ToggleFF_next_ka);
+        } else {
+            SETCALC(ToggleFF_next);
+        }
+    }
 
     unit->m_prevtrig = 0.f;
     unit->mLevel = 0.f;
 
-    ZOUT0(0) = 0.f;
+    ToggleFF_next(unit, 1);
+    // no need to reset internal state
+    // if trig is detected in Ctor, UGen doesn't need to toggle again in _next
 }
 
-
 void ToggleFF_next(ToggleFF* unit, int inNumSamples) {
     float* out = ZOUT(0);
     float* trig = ZIN(0);
     float prevtrig = unit->m_prevtrig;
     float level = unit->mLevel;
 
     LOOP1(inNumSamples, float curtrig = ZXP(trig); if (prevtrig <= 0.f && curtrig > 0.f) level = 1.f - level;
-          ZXP(out) = level; prevtrig = curtrig;);
+          prevtrig = curtrig; ZXP(out) = level;);
```

Comment:
Any particular reason to rearrange these? In general we'd like to keep non-functional changes to a minimum. 

And I think generally, setting state variables after the output is a common pattern, as it's prepping that variable for the next round of calculations.

Reply:
thanks! It was a leftover from my previous implementation